File: apigateway_getting_started/create_api_client_resource_get.py
import botocore
import boto3
import logging

logger = logging.getLogger()
apig = boto3.client("apigateway")

try:
    result = apig.put_method(
        restApiId="abcd1234",
        resourceId="aaa111",
        httpMethod="GET",
        authorizationType="NONE",
    )
except botocore.exceptions.ClientError as error:
    logger.exception("The 'GET /pets' method setup failed: %s", error)
    raise
attribute = ["httpMethod", "authorizationType", "apiKeyRequired"]
filtered_result = {key: result[key] for key in attribute}
print(filtered_result)

File: apigateway_getting_started/create_api_client_resource_get.js
import { APIGatewayClient, PutMethodCommand } from "@aws-sdk/client-api-gateway";
(async function () {
    const apig = new APIGatewayClient({ region: "us-east-1" });
    const command = new PutMethodCommand({
        restApiId: 'abcd1234',
        resourceId: 'aaa111',
        httpMethod: 'GET',
        authorizationType: 'NONE'
    });
    try {
        const results = await apig.send(command)
        console.log(results)
    } catch (err) {
        console.log("The 'GET /pets' method setup failed:\n", err)
    }
})();